### PR TITLE
feat: Add auto-generated warning to Markdown output

### LINKED ISSUE #309: Optionally add an "autogenerated" warning comment in markdown mode
**Is your feature request related to a problem? Please describe.**

Sometimes, folks working on a repo don't realize that the README is generated by terraform-docs, and they edit the file in place. 

**Describe the solution you'd like**

In markdown mode, I'd like `terraform-docs` to (optionally) add a comment to the top of the generated doc, like

```
<!-- Autogenerated by terraform-docs. Do not modify this file directly. -->
```

(or something user-specifiable, so that it's not limited to English-speakers maybe.)

**Describe alternatives you've considered**

Any other approach means putting the comment in the _source_, which defeats the purpose, because people should modify the source!

### PR DESCRIPTION
**Pull request by alexjurkiewicz**
Fixes #309.

### Prerequisites

Put an `x` into the box(es) that apply:

- [ ] This pull request fixes a bug.
- [x] This pull request adds a feature.
- [x] This pull request enhances existing functionality.
- [ ] This pull request introduces breaking change.

For more information, see the [Contributing Guide](https://github.com/terraform-docs/terraform-docs/tree/master/CONTRIBUTING.md).

### Description

Adds the following header to Markdown output:

```markdown
<!-- This file was auto-generated by terraform-docs -->
```

### Issues Resolved

#309 

### Checklist

Put an `x` into all boxes that apply:

- [x] I have read the [Contributing Guidelines](https://github.com/terraform-docs/terraform-docs/tree/master/CONTRIBUTING.md).

#### Tests

- [ ] I have added tests to cover my changes.
- [ ] All tests pass when I run `make test`.

#### Documentation

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

#### Code Style

- [x] My code follows the code style of this project.

### CODE REVIEW

**Comment by alexjurkiewicz** [comment]
I'm also interested in working on a feature `--footer-from` which is like `--header-from`. Are you interested in such a contribution?

**Review by khos2ow — COMMENTED** [review]
@alexjurkiewicz thank you for your contribution and I apologize for massive delay in response.

Unfortunately we cannot proceed with this PR as it stands. Would you still be willing to work on it? If yes, please rebase on master and we can take it from there.

**Comment by alexjurkiewicz** [comment]
done!

File: internal/format/markdown_document.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ import (
 
 const (
 	documentHeaderTpl = `
+	<!-- This file was auto-generated by terraform-docs -->
```
Comment (khos2ow) [inline internal/format/markdown_document.go:23]: Please move this line at the beginning of `documentTpl`.

File: internal/format/markdown_table.go [hunk 2]
```diff
@@ -20,6 +20,8 @@ import (
 
 const (
 	tableHeaderTpl = `
+	<!-- This file was auto-generated by terraform-docs -->
```
Comment (khos2ow) [inline internal/format/markdown_table.go:23]: The same as above.

**Review by khos2ow — CHANGES_REQUESTED** [review]
Awesome, thanks. Please also note that, since a little while ago, we require for all the commits to be signed (please refer to [contributing guide](https://github.com/terraform-docs/terraform-docs/blob/master/CONTRIBUTING.md#contributing-code) for more detail).

There are few items I'd like to mention:

- couple comments below
- please add similar comment for AsciiDoc files too
- please update test golden files accordingly

**Comment by khos2ow** [comment]
Also I'm thinking maybe it's better to rephrase the comment as the original comment in #309, which has some emphasis on `do not modify`.

**Comment by alexjurkiewicz** [comment]
Sorry, doesn't look like I can generate signed commits via the Github web UI. I hope you can pick this up yourself 🙏

**Comment by khos2ow** [comment]
Sorry @alexjurkiewicz, my bad. I should've been more clear. We require commits to be signed-off (`git commit -s`) and not GPG signed.

**Comment by khos2ow** [comment]
Closing this in favor of https://github.com/terraform-docs/terraform-docs/issues/309#issuecomment-797045473

**Pull request closed without merging**